substratest/assets.py

```diff
@@ -21,9 +21,12 @@ def get(self):
         raise NotImplementedError
 
 
-class TupleStatus(Enum):
-    done = auto()
-    failed = auto()
+class TupleStatus(enum.Enum):
+    doing = enum.auto()
+    done = enum.auto()
+    failed = enum.auto()
+    todo = enum.auto()
+    waiting = enum.auto()
```

Comment:
Very naive question here for @samlesu (since you suggested an enum) : what do we get out of an enum that we don't by using a basic class with string properties, something like:
```python
class TupleStatus:
   doing = 'doing'
```

In this specific case, using an enum creates a bit of overhead everywhere : we cant do `assert traintuple['status'] == TupleStatus.done`, we need to use `TupleStatus.done.name` which seems to defeat the purpose of using an enum in the first place.

Reply:
Good question !

I do agree that having to use the `name` property of an enum to check equality adds a layer of complexity. This could be improved by setting an enum in the `Asset` instance instead of a string. We should not forget that types defined in the dataclass are for "linting" purposes, there is no validation.

Right now, I don't see any advantage of using an enum rather than class properties. Till we don't have validation, using class properties seems like a good alternative solution.